src/opentimelineio/clip.cpp

```diff
@@ -19,33 +25,74 @@ Clip::~Clip()
 MediaReference*
 Clip::media_reference() const noexcept
 {
-    return _media_reference;
+    auto active = _media_references.find(_active_media_reference);
+    return active == _media_references.end() || !active->second
+               ? nullptr
+               : active->second;
+}
+
+Clip::MediaReferences
+Clip::media_references() const noexcept
+{
+    MediaReferences result;
+    for (auto const& m: _media_references)
+    {
+        result.insert(
+            { m.first, dynamic_retainer_cast<MediaReference>(m.second) });
+    }
+    return result;
+}
+
+void
+Clip::set_media_references(MediaReferences const& media_references) noexcept
```

Comment:
There exists potential here to set a map where none of the keys match the `active_reference`.  Currently, if this is the case when they call `media_reference`, it will return a MissingReference. Is that sufficient for error handling? Or do we want to take an ErrorStatus parameter here like some other methods do? Or throw an exception?

Reply:
Checking the current behavior of `Clip` in python, it appears that setting `media_reference` to `None` results in it being set to a blank `MissingReference` - so I think returning `MissingReference` feels pretty consistent with our current behavior.

Reply:
I think set_media_reference has to clear the active_reference. The reason is that we are REPLACING active media_references, not merging them. So if before/after maps both have a "fizzbin" tag, and "fizzbin" was the active reference, the correspondence is superficial. 

In general, setting the media references, and thereby introducing a potential consistency problem (before has "fizzbin", after does not), seems like an undesirable behavior.

Reply:
I think you mean `set_media_references` (plural, not singular, correct?). 

Assuming that's the case, we could potentially modify the signature to `set_media_references( MediaReferences, std::string);` forcing the caller to tell us which reference in the new map is the active one.